HowTo/gRPC/Go/use_gRPC_with_go.md

```diff
@@ -0,0 +1,249 @@
+In order to prepare our gRPC server application to deploy to App Service, we will need to configure the Go app with an additional port that so that the system can ping the application for Healthcheck requests.
+
+In your **main.go** add the following code to configure the extra port.  
+
+```Go
+// Configure Go to listen Healthcheck requests 
+func async[T any](f func() T) chan T {
+    ch := make(chan T)
+    go func() {
+        ch <- f()
+    }()
+    return ch
+}
+
+func HttpServer() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe(":8080", nil)
+}
+
+func main() {
+    async(func() string {
+		HttpServer()
+		return "1"
+    })
+.....
+}
+
+```
+
+Once configured this will ensure that your application is listening to a specific HTTP only port, which will be needed when we deploy to App Service. 
+
+This is the code of your completed **main.go**.
+
+```Go
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"google.golang.org/grpc"
+	pb "helloworld/helloworld"
+)
+
+var (
+	port = flag.Int("port", 50051, "The server port")
+)
+
+// server is used to implement helloworld.GreeterServer.
+type server struct {
+	pb.UnimplementedGreeterServer
+}
+
+// SayHello implements helloworld.GreeterServer
+func (s *server) SayHello(ctx context.Context, in *pb.HelloRequest) (*pb.HelloReply, error) {
+	log.Printf("Received: %v", in.GetName())
+	return &pb.HelloReply{Message: "Hello " + in.GetName()}, nil
+}
+
+func HelloServer(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintf(w, "Hello World")
+}
+
+func async[T any](f func() T) chan T {
+    ch := make(chan T)
+    go func() {
+        ch <- f()
+    }()
+    return ch
+}
+
+func HttpServer() {
+	http.HandleFunc("/", HelloServer)
+	http.ListenAndServe(":8080", nil)
+}
+
+func main() {
+    async(func() string {
+		HttpServer()
+		return "1"
+    })
+	flag.Parse()
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+	s := grpc.NewServer()
+	pb.RegisterGreeterServer(s, &server{})
+	log.Printf("server listening at %v", lis.Addr())
+	if err := s.Serve(lis); err != nil {
+		log.Fatalf("failed to serve: %v", err)
+	}
+}
+
+
+```
+You can run your code locally to ensure that it works. Once this is done your application is now ready to deploy to App Service.
+
+### Deploy to App Service
+Now that you have your server application setup and running locally, you can go to the portal and create your web app.  One thing to note is that Go and gRPC is currently only supported on Linux so be sure to choose this option when creating your web app.
+
+Create your web app as you normally would.  Choose **Code** as your Publish option.  Choose **Go 1.18 (Experimental)** or **Go 1.19 (Experimental)** as your Runtime stack and **Linux** as your Operating System.  
+
+Now that your web app is created, you'll need to do the following before deploying your application:
+
+#### 1. Enable HTTP version
```

Comment:
@tulikac This should be *Set HTTP version*

Reply:
Done